src/backend/distributed/commands/dependencies.c

```diff
@@ -135,6 +143,95 @@ EnsureDependenciesExistOnAllNodes(const ObjectAddress *target)
 }
 
 
+/*
+ * EnsureDependenciesCanBeDistributed ensures all dependencies of the given object
+ * can be distributed.
+ */
+static void
+EnsureDependenciesCanBeDistributed(const ObjectAddress *objectAddress)
+{
+	/* If an object circularcly depends to itself, Citus can not handle it */
+	ErrorIfCircularDependencyExists(objectAddress);
+
+	/* If any of the dependency of the object can not be distributed, error out */
+	ObjectAddress *undistributableDependency = GetUndistributableDependency(
```

Comment:
2 questions:
1) Do we still need the other places we call `GetUndistributableDependency`? Isn't this going to cover all?  I guess we'll regress a bit on the error message but seems fine to me. What is your take?

2) Seems like a good time to move this logic to a function name: `ErrorIfHasUnsupportedDependency` or such

Reply:
1. I'm inclined to keep them as we are using GetUndistributableDependency to decide whether we will create function/aggregates locally on other places.
2. Done

Reply:
1. We can use DeferredError infrastructure? Instead of throwing errors, we DeferrredErrors such as [here](https://github.com/citusdata/citus/blob/52879fdc96bf97aa5cdeea276eab2969f19cbef4/src/backend/distributed/planner/query_pushdown_planning.c#L599)?

My point is that it'd be good to keep this logic centralized given its importance and user facing impact

Reply:
Hmm DeferredError makes sense, let me check and update the PR accordingly.